lib/eslint.js

```diff
@@ -1036,8 +1036,8 @@ module.exports = (function() {
 
             // if current node is function declaration, add it to the list
             var current = controller.current();
-            if (["FunctionDeclaration", "FunctionExpression",
-                    "ArrowFunctionExpression", "SwitchStatement"].indexOf(current.type) >= 0) {
+            if (["FunctionDeclaration", "FunctionExpression", "ArrowFunctionExpression"].indexOf(current.type) >= 0 ||
+                    currentConfig.ecmaFeatures.blockBindings) {
```

Comment:
Yes, I was letting go some "false positives" for escope to catch. I think now it's fixed.